src/components/proposed-date-label/index.tsx

```diff
@@ -21,21 +21,25 @@ import useMeta from '../../hooks/use-meta';
  */
 export const ProposedDateLabel: FC<{
 	date?: string;
-	isFloating: boolean;
+	postStatus: string;
 	proposedDate?: string;
-}> = ( { date, isFloating, proposedDate } ) => {
+}> = ( { date, postStatus, proposedDate } ) => {
 	const { dateFormat } = useExperimentalSettings( ( settings ) => ( {
 		dateFormat: `${ settings.formats.date } ${ settings.formats.time }`,
 	} ) );
 
-	if ( date && ! isFloating ) {
-		return dateI18n( dateFormat, date );
+	if ( postStatus !== 'draft' ) {
+		return __( 'Immediately' );
```

Comment:
Why should this be the default for all non-draft states?

Reply:
The original intent of this code was to let a set `date` take priority over the presence of a meta date. This UI change implies that is not the case

Reply:
So the Use Case is - If date is set one time, then requestor can not change the date to reschedule the post, that's Why Added the precedence  for `proposedDate`.

Reply:
And I guess we can get rid of
```
if ( postStatus !== 'draft' ) {
		return __( 'Immediately' );
}
```

Reply:
My best understanding right now is that the repo where we are using this includes logic with _forces_ the proposed date to override a set `post_date`, whereas the plugin does not account for that. Once a post_date has been set, the default logic inside `namespace.php` says,

```php
	// Determine whether the circumstances warrant applying the proposed date.
	$accept_proposal = (
		// Do not override date on a post that already had an explicit date.
		status_has_floating_date( $old_status )
	) && (
		// No update is needed at this time if new status is still "date floating".
		! status_has_floating_date( $data['post_status'] )
	) && (
		// Post must not yet be scheduled.
		$existing_post->post_date_gmt === '0000-00-00 00:00:00'
	);
```

That last clause means that the plugin doesn't currently account on a PHP level for changing the date of a previously-scheduled post. This is why I'm arguing against changing this to make the proposed date display even if there's a post date present. I do not think that is how this plugin would behave out of the box in a different environment than the one where you're testing it.

I will validate that assumption later today, though; I know I'm holding you up on this!